src/python/pants/backend/python/macros/pants_requirements_test.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import pytest
+from packaging.version import Version
+
+from pants.backend.python.macros import pants_requirements
+from pants.backend.python.macros.pants_requirements import (
+    GenerateFromPantsRequirementsRequest,
+    PantsRequirementsTargetGenerator,
+    determine_version,
+)
+from pants.backend.python.pip_requirement import PipRequirement
+from pants.backend.python.target_types import PythonRequirementModulesField, PythonRequirementsField
+from pants.engine.addresses import Address
+from pants.engine.target import GeneratedTargets
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.mark.parametrize(
+    "pants_version,expected",
+    (
+        ("2.4.0.dev1", "==2.4.0.dev1"),
+        ("2.4.0rc1", ">=2.4,<2.5"),
```

Comment:
Not sure if this should be `>=2.4.0rc0` instead of `>=2.4`? I don't think it should be `>=2.4.0rc1` because that would imply that when you bump an rc or stable release, you force all consumers to also make the same bump.